tests/test_grids.py

```diff
@@ -601,3 +601,128 @@ class MyParticle(ptype[mode]):
         assert np.allclose(pset.meridional, 0.01)
         assert np.allclose(pset.vert, -0.01)
         assert np.allclose(pset.tracer, 1)
+
+
+@pytest.mark.parametrize('mode', ['scipy', 'jit'])
+@pytest.mark.parametrize('gridindexingtype', ['mitgcm', 'nemo'])
+def test_mitgridindexing(mode, gridindexingtype):
+    xdim, ydim = 151, 201
+    a = b = 20000  # domain size
+    lon = np.linspace(-a / 2, a / 2, xdim, dtype=np.float32)
+    lat = np.linspace(-b / 2, b / 2, ydim, dtype=np.float32)
+    dx, dy = lon[2] - lon[1], lat[2] - lat[1]
+    omega = 2 * np.pi / delta(days=1).total_seconds()
+
+    index_signs = {'nemo': -1, 'mitgcm': 1}
+    isign = index_signs[gridindexingtype]
+
+    def calc_r_phi(ln, lt):
+        return np.sqrt(ln ** 2 + lt ** 2), np.arctan2(ln, lt)
+
+    def calculate_UVR(lat, lon, dx, dy, omega):
+        U = np.zeros((lat.size, lon.size), dtype=np.float32)
+        V = np.zeros((lat.size, lon.size), dtype=np.float32)
+        R = np.zeros((lat.size, lon.size), dtype=np.float32)
+        for i in range(lon.size):
+            for j in range(lat.size):
+                r, phi = calc_r_phi(lon[i], lat[j])
+                R[j, i] = r
+                r, phi = calc_r_phi(lon[i] + isign * dx / 2, lat[j])
+                V[j, i] = -omega * r * np.sin(phi)
+                r, phi = calc_r_phi(lon[i], lat[j] + isign * dy / 2)
+                U[j, i] = omega * r * np.cos(phi)
+        return U, V, R
+
+    U, V, R = calculate_UVR(lat, lon, dx, dy, omega)
+    data = {'U': U, 'V': V, 'R': R}
+    dimensions = {'lon': lon, 'lat': lat}
+    fieldset = FieldSet.from_data(data, dimensions, mesh='flat', gridindexingtype=gridindexingtype)
+    fieldset.U.interp_method = 'cgrid_velocity'
+    fieldset.V.interp_method = 'cgrid_velocity'
```

Comment:
interesting - I still didn't figure out how the field loading works in that case, but I guess that your fixes really just had an influence on the indexing and on nothing else, so nothing had to be done with the FileBuffer .. that would be quite cool

Reply:
In this case, it's indeed only indexing. When using MITgcm, there's multiple ways to output netCDF data. By default, output is in its native binary format. This can be converted to netCDF with the [`xmitgcm`](https://github.com/MITgcm/xmitgcm) package. Alternatively, output can be written to netCDF directly with the `mnc` or the `NF90IO` packages. `xmitgcm` outputs a velocity field with similar shapes as NEMO (UVEL[nx+1, ny+1]). Here U is located at the inflow coordinates instead, and this was easy to implement (credits to @erikvansebille and @SLYpma). Using `mnc` or `NF90IO` is where the trouble starts: dimensions will in that case correspond to the exact number of U or V faces (so UVEL[nx+1, ny], VVEL[nx, ny+1]). Turned out to pretty complicated to rewrite Parcels to accept fields with uneven dimensions in a C-grid, so for now those are not supported.